aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -1343,24 +1363,33 @@ func resourceAviatrixTransitGatewayRead(d *schema.ResourceData, meta interface{}
 			ResourceName: d.Get("gw_name").(string),
 			CloudType:    gw.CloudType,
 		}
-
-		tagList, err := client.GetTags(tags)
+		err := client.GetTagsMap(tags)
 		if err != nil {
-			return fmt.Errorf("unable to read tag_list for gateway: %v due to %v", gateway.GwName, err)
+			return fmt.Errorf("unable to read tags for transit gateway: %v due to %v", gateway.GwName, err)
 		}
 
-		var tagListStr []string
 		if _, ok := d.GetOk("tag_list"); ok {
+			tagList := make([]string, 0, len(tags.Tags))
+			for key, val := range tags.Tags {
+				str := key + ":" + val
+				tagList = append(tagList, str)
+			}
+
 			tagList1 := d.Get("tag_list").([]interface{})
```

Comment:
Can we think of a better name than `tagList1`? Maybe `tagsFromUserConfig` or something more descriptive